src/server.py

```diff
@@ -77,7 +79,8 @@ async def handle_client(reader, writer):
         # Handle user authentication
         auth_data = await reader.read(1024)
         username, password = auth_data.decode().strip().split(':')
-        user_id = await register_user(username, password)
+        hashed_password = hashlib.sha256(password.encode()).hexdigest()
```

Comment:
I'm not particularly into using `SHA-256`. Although it's a solid algorithm, `bcrypt` outperforms it in terms of anti-brute-force protection.

<img width="746" alt="Screenshot 2024-08-22 at 21 12 36" src="https://github.com/user-attachments/assets/689e57ff-3760-4ed1-b570-a68a9be2057c">